src/aws-util/src/util.rs

```diff
@@ -11,6 +11,8 @@ use aws_smithy_client::erase::DynConnector;
 use aws_smithy_client::hyper_ext::Adapter;
 use hyper_tls::HttpsConnector;
 
+// If no `aws-sdk-*` feature is enabled, this function appears unused.
+#[allow(dead_code)]
```

Comment:
The convention for doing this is to use `#[cfg(feature = ...)]` attributes to conditionally exclude code that is not active for a particular set of features. A bit more precise than turning of dead code analysis entirely.

In this case I'd recommend attaching that attribute to the `util` mode as a whole!

```diff
diff --git a/src/aws-util/src/lib.rs b/src/aws-util/src/lib.rs
index afeabf79c..f01dcfe38 100644
--- a/src/aws-util/src/lib.rs
+++ b/src/aws-util/src/lib.rs
@@ -12,6 +12,7 @@
 #![warn(missing_docs, missing_debug_implementations)]
 #![cfg_attr(nightly_doc_features, feature(doc_cfg))]
 
+#[cfg(any(feature = "kinesis", feature = "sqs", feature = "s3", feature = "sts"))]
 mod util;
 
 pub mod config;
```

Reply:
Thanks, my reasoning was that it might become a hassle to maintain that `cfg` block as AWS resources are added or features renamed. But in hindsight, it's probably not likely that this happens very often.

Reply:
Yeah, you'll be in lib.rs to add the new module conditionally anyway, so should hopefully be obvious that you need to update the attribute on `util` too!